diesel/src/result.rs

```diff
@@ -287,6 +301,13 @@ impl<T> OptionalExtension<T> for QueryResult<T> {
             Err(e) => Err(e),
         }
     }
+    fn optional_empty_changeset(self) -> Result<Option<T>, Error> {
```

Comment:
I just used the same trait here, hope that's okay.

Reply:
I fear that this is a breaking change, as someone could implement `OptionalExtension` for their own types, which in turn would then not implement the new method.

(I doubt that anyone will do this, but better be safe than sorry)

Reply:
No problem, did this now.